src/set_theory/game/nim.lean

```diff
@@ -180,9 +180,9 @@ begin
     { exact ordinal.le_total O₁ O₂ },
     { have h : O₁ < O₂ := lt_of_le_of_ne h' h,
       rw [impartial.first_wins_symm', lf_def_le, nim_def O₂],
-      refine or.inl ⟨(left_moves_add (nim O₁) _).symm (sum.inr _), _⟩,
-      { exact (ordinal.principal_seg_out h).top },
-      { simpa using (impartial.add_self (nim O₁)).2 } },
+      refine or.inl
+        ⟨@to_left_moves_add _ ⟨_, _, _, _⟩ (sum.inr (ordinal.principal_seg_out h).top), _⟩,
```

Comment:
Needing to introduce an `@` is always slightly worrying. Could you explain why it is necessary here, if it really is?

Reply:
The reason `@` is needed here is because we need to give Lean the hint that we're calling it on a move of a game `mk _ _ _ _`. The purpose of this new API is to precisely avoid the situations where we need to unfold our games in order to talk about addition, so I expect this circumstance to be rare moving forward.

This is somewhat outside of the scope of this PR, but I could attempt to rewrite this proof to see if this `@` could be avoided.

Reply:
(note that a lot of the nim API currently does a lot of unfolding and "abuses" def-eq)